pkg/autoscaler/autoscaler.go

```diff
@@ -227,6 +214,8 @@ func (a *Autoscaler) autoscaleFn(api userconfig.Resource) (func() error, error)
 			recommendation = autoscalingSpec.MaxReplicas
 		}
 
+		recs := a.recs[api.Name]
```

Comment:
nit: Is this line necessary or just for convenience? Also, if just for convenience, would it make sense to move it above this function (e.g. next to `a.recs[api.Name] = newRecommendations()`)?